pkg/smartcontract/param_context.go

```diff
@@ -40,6 +40,53 @@ func NewParameter(t ParameterType) Parameter {
 	}
 }
 
+type keyValuePair struct {
+	Key   interface{} `json:"key"`
+	Value Parameter   `json:"value"`
+}
+
+// MarshalJSON implements Marshaler interface.
+func (p *Parameter) MarshalJSON() ([]byte, error) {
+	switch p.Type {
+	case BoolType, IntegerType, PublicKeyType, StringType, Hash256Type, Hash160Type:
+		return json.Marshal(Parameter{
+			Type:  p.Type,
+			Value: p.Value,
+		})
+	case SignatureType, ByteArrayType:
+		return json.Marshal(Parameter{
+			Type:  p.Type,
+			Value: hex.EncodeToString(p.Value.([]byte)),
+		})
+	case ArrayType:
+		var value = Parameters{}
```

Comment:
value := make([]rawParameter, 0)
paramz := p.Value.([]Parameter)
for i := range paramz {
rj, err := json.Marshal(&paramz[i])
value = append(value, rj)
}
rj, err := json.Marshal(value)